### ISSUE TITLE
property-whitelist issues

### ISSUE THREAD

**Issue by ntwb** [stylelint/stylelint#445]
Given the following rule configuration: `"property-whitelist": [ 2, "margin" ],` with this CSS:

```css
.test {
	argin: 0;
}
```

I'd expect this to show a warning: `Unexpected property "argin" (property-whitelist)` and it doesn't.

A few more CSS examples other than `argin` where no warning is shown are `rgin`, `gin`, `in`, `n` :smirk:

Using `yolo: 0;` correctly warns with: `Unexpected property "yolo" (property-whitelist)`
Using `www: 0;` correctly warns with: `Unexpected property "www" (property-whitelist)`
Using `mmargin: 0;` correctly warns with: `Unexpected property "mmargin" (property-whitelist)`
Using `marginn: 0;` correctly warns with: `Unexpected property "marginn" (property-whitelist)`
Using `arin: 0;` correctly warns with: `Unexpected property "arin" (property-whitelist)`

I'm guessing it's a regex issue somewhere in that "any" set of characters in the same order as the string contained in the option value are ignored/skipped and not detected as invalid.

If it's not obvious, I'm trying to detect a typo of the word `margin` :wink:

**Comment by jeddy3** [comment]
Thanks for the report :)

Is `property-whitelist": [ 2, "margin" ]` exactly what you've used in the config? If it is, can you try `property-whitelist": [ 2, ["margin"]]` instead?

The rule [expects](https://github.com/stylelint/stylelint/blob/master/src/rules/property-whitelist/README.md) an `array` of property names, not a single `string`.

So, I think you've uncovered a different bug i.e. it should warned you about the misconfigured rule but, and I can replicate this locally, it currently does not

@davidtheclark What should the `possible` value be [here](https://github.com/stylelint/stylelint/blob/master/src/rules/property-whitelist/index.js#L19) if we wish to accept a one-level deep array of arbitrary size and arbitrary `strings`?

**Comment by davidtheclark** [comment]
There is not currently a way to check that the option is itself an array. It was built to check every item in the array. This will take a little time to change correctly. In the meantime, I think that @jeddy3's suggestion (to use `["margin"]` instead of `"margin"`) might do the trick.

BTW: I am going to start dedicating most of my freetime over the next month to preparing my talk for http://cssday.io/. So although I will try to pitch in for bug fixes I probably will not spend much time developing new features, for a little while.

**Comment by jeddy3** [comment]
> to preparing my talk for http://cssday.io/.

Good luck :)

@ntwb Let me know if using `["margin']` resolves your immediate issue. I'll then close this and create a new issue to capture of the options stuff.

**Comment by ntwb** [comment]
> Is `property-whitelist": [ 2, "margin" ]` exactly what you've used in the config?

Yes

> If it is, can you try `property-whitelist": [ 2, ["margin"]]` instead?

Yes, using this it works as expected and correctly reports a warning:
• `Unexpected property "argin" (property-whitelist) [stylelint]`

> I think that @jeddy3's suggestion (to use ["margin"] instead of "margin") might do the trick.

Indeed :+1: 

> @ntwb Let me know if using ["margin'] resolves your immediate issue. I'll then close this and create a new issue to capture of the options stuff.

Yes, resolved, thanks :+1: 

> preparing my talk for http://cssday.io/
Likewise, good luck, I'd love to come and see your talk, but ~8,500 miles from home for me :smirk:
